Automattic/newspack-popups
# feat: add segmentation README and more useful info to debug object

### All Submissions:






### Changes proposed in this Pull Request:



Adds more useful debugging info to the front-end segmentation API. Instead of just telling you whether or not each prompt is displayed the debug object now includes the following info for each prompt:

- `displayed`: Whether or not the prompt was displayed
- `element`: A live reference to the HTMLElement of the prompt
- `override`: If the prompt was forced to be displayed or suppressed via override, this will show the boolean value of the override.
- `suppression`: If the prompt was not displayed, this will be an array containing all the reasons it was suppressed.

This PR also allows the debug object to be rendered if the `NEWSPACK_POPUPS_DEBUG` constant is defined and truthy or the `NEWSPACK_LOG_LEVEL` constant is defined and greater than `1`, which allows for debugging Campaigns segmentation without turning on `WP_DEBUG` for the whole site.

Also adds a README.md to explain some technical details on how the new segmentation API works.

### How to test the changes in this Pull Request:

1. With `WP_DEBUG` or `NEWSPACK_POPUPS_DEBUG` defined as `true`,  or `NEWSPACK_LOG_LEVEL` defined as `2` or greater, view a page in a new session.
2. In the JS console, look for the `newspack_popups_debug` object. Confirm that it appears with debugging info for each prompt as described above, and that the debugging info is accurate for each prompt.
3. Remove all the constants from step 1, refresh, and confirm that `newspack_popups_debug` is not defined.

### Other information:
